pkg/explorer/baton_service.go

```diff
@@ -0,0 +1,449 @@
+package explorer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/conductorone/baton-sdk/pkg/connectorstore"
+	"github.com/conductorone/baton-sdk/pkg/dotc1z/manager"
+	v1 "github.com/conductorone/baton/pb/baton/v1"
+	"github.com/conductorone/baton/pkg/storecache"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	reader_v2 "github.com/conductorone/baton-sdk/pb/c1/reader/v2"
+)
+
+type BatonService struct {
+	filePath     string
+	syncID       string
+	resourceType string
+}
+
+func (b BatonService) GetEntitlements(ctx context.Context) (*v1.EntitlementListOutput, error) {
+	c1zPath := b.filePath
+
+	m, err := manager.New(ctx, c1zPath)
```

Comment:
We'll want to load the c1z and create the store cache once per baton service. This will save a lot of effort, especially with larger c1zs, or when they are loaded from s3.

Reply:
yeah makes sense, thank you!